promotion/claim.go

```diff
@@ -116,7 +118,28 @@ func (service *Service) ClaimPromotionForWallet(
 
 	walletIsReputable, err := service.reputationClient.IsWalletReputable(ctx, walletID)
 	if err != nil {
-		return nil, err
+		responseError, ok := err.(errs.ResponseError)
+		if !ok {
+			return nil, errors.New("An unknown error occured")
+		}
+		switch responseError.Cause() {
```

Comment:
is this just a demo of how one could check the error Cause / Status? just confused because we don't actually do anything based on the conditions here

Reply:
I think it would be good to instead demonstrate how one can bubble up an error and change the HTTP response depending upon the cause

Reply:
yup, just a demo. wanted to get your feedback on the error coming out of the reputation file or the error definition itself. if you had other ideas for it.